pymc3/ode/ode.py

```diff
@@ -0,0 +1,170 @@
+
+    .. code-block:: python
+
+        def odefunc(y,t,p):
+            #Logistic differential equation
+            return p[0]*y[0]*(1-y[0])
+        
+        times = np.arange(0.5, 5, 0.5)
+        
+        ode_model = DifferentialEquation(func = odefunc, t0 = 0, times = times, n_states = 1, n_odeparams = 1)
+    '''
+
+    __props__ = ()
+    
+    def __init__(self, func, times, n_states, n_odeparams, t0=0):
+        if not callable(func):
+            raise ValueError("Argument func must be callable.")
+        if n_states<1:
+            raise ValueError('Argument n_states must be at least 1.')
+        if n_odeparams<0:
+            raise ValueError('Argument n_states must be non-negative.')
+
+        #Public
+        self.func = func
+        self.t0 = t0
+        self.times = times
+        self.n_states = n_states
+        self.n_odeparams = n_odeparams
+
+        #Private
+        self._n = n_states
+        self._m = n_odeparams + n_states
+
+        self._augmented_times = np.insert(times, t0, 0)
+        self._augmented_func = augment_system(func, self._n, self._m)
+        self._sens_ic = self._make_sens_ic()
+
+        self._cached_y = None
+        self._cached_sens = None
+        self._cached_parameters = None
+
+        self._grad_op = ODEGradop(self.numpy_vsp)
+
+
+    def _make_sens_ic(self):
+        # The sensitivity matrix will always have consistent form.
+        # If the first n_odeparams entries of the parameters vector in the simulate call
+        # correspond to ode paramaters, then the first n_odeparams columns in
+        # the sensitivity matrix will be 0
+        sens_matrix = np.zeros((self._n, self._m))
+
+        # If the last n_states entrues of the paramters vector in the simulate call
+        # correspond to initial conditions of the system,
+        # then the last n_states columns of the sensitivity matrix should form
+        # an identity matrix
+        sens_matrix[:, -self.n_states:] = np.eye(self.n_states)
+
+        # We need the sensitivity matrix to be a vector (see augmented_function)
+        # Ravel and return
+        dydp = sens_matrix.ravel()
+
+        return dydp
+
+    def _system(self, Y, t, p):
+        '''
+        This is the function that will be passed to odeint.
+        Solves both ODE and sensitivities
+        Args:
+            Y (vector): current state and current gradient state
+            t (scalar): current time
+            p (vector): parameters
+        Returns:
+            derivatives (vector): derivatives of state and gradient
+        '''
+
+        dydt, ddt_dydp = self._augmented_func(Y[:self._n], t, p, Y[self._n:])
+        derivatives = np.concatenate([dydt, ddt_dydp])
+        return derivatives
+
+    def _simulate(self, parameters):
+        # Initial condition comprised of state initial conditions and raveled
+        # sensitivity matrix
+        y0 = np.concatenate([ parameters[self.n_odeparams:] , self._sens_ic])

```

Comment:
Trailing whitespace after `[`

Reply:
Our pylint settings are actually quite permissive when it comes to spaces. There are several empty lines, or spaces missing around operators